### ISSUE TITLE
IDR - Gene Tree - Mockups

### ISSUE THREAD

**Issue by gusferguson** [openmicroscopy/design#23]
As discussed on https://trello.com/c/s0HD7Vi6/173-tagexport-py

Requirement is to be able to see the genes found in metadata across screens.
- need to be able to filter according to gene symbol
- restrict to selected species

Also suggestion that it would be good to be able to opt to see similar genes or orthologs. No obvious way to do that other than by using wildcards in filter. To be discussed later.

Mockup shows use of filter.
Only show gene symbols, not identifiers.

![ir gene tree 1](https://cloud.githubusercontent.com/assets/1552703/13640904/672c8844-e60e-11e5-8471-d53d7b402288.png)

Drop down shows species available

![ir gene tree - dd](https://cloud.githubusercontent.com/assets/1552703/13640912/73e464a8-e60e-11e5-957a-3153e9fcc8fe.png)

**Comment by joshmoore** [comment]
Thanks, @gusferguson. One looming question for me is, what appears in the right and even middle panels when the gene itself is clicked on. This brings us back to the folders discussion from today.

cc: @aleksandra-tarkowska @sbesson @dominikl @mtbc

**Comment by gusferguson** [comment]
@joshmoore 

> One looming question for me is, what appears in the right and even middle panels when the gene itself is clicked on. This brings us back to the folders discussion from today.

Probably need to show all images in the centre pane - nightmare if very large - maybe just give first n and say there are x in total.

**Comment by gusferguson** [comment]
Added RFEs to issue above.

**Comment by jburel** [comment]
I will assume that the "gene" will be configurable otherwise we will have a growing number of them.
cf. mock-up on ROI

**Comment by gusferguson** [comment]
Will add in some options for selection of genes and clarify what the counts are (when it is clear to me! :-).

**Comment by joshmoore** [comment]
@jburel : initially saw as a plugin, but yeah, if they don't somehow scroll/collapse it could overgrow the available space. Another option would be to more generally define these as "Hierarchies" (trees/folders/... here we go again ;) ) and show "Gene" as another type that is a "root" (cc @sbesson from discussions yesterday)

**Comment by aleksandra-tarkowska** [comment]
> >    One looming question for me is, what appears in the right and even middle panels when the gene itself is clicked on. This brings us back to the folders discussion from today.

> Probably need to show all images in the centre pane - nightmare if very large - maybe just give first n and say there are x in total.

pagination seems to work nice, just need a better styling for pages as they cover too much of the centre panel

**Comment by eleanorwilliams** [comment]
Organism.  The organism/species used in a screen is not included in the bulk annotation or map annotations as it is usually/always the same for all cells in a screen and would to included it would mean a lot of repeated information.   But we could put this information in a map annotation at the screen level to include this.    

We need a copy of the idr-demo to add this, as the current demo is read-only so that we don't have to worry about backups/versions.

**Comment by eleanorwilliams** [comment]
I think it would be better to show all the images relating to one well as a 'well' e.g. in mitocheck there are images from many timepoints and it would be good to show them as one well rather than each individual image.

**Comment by eleanorwilliams** [comment]
Plates - do we need to show the plate level in the hierarchy?  I am not sure about this.  In Mitocheck, the plate names are included in the well level information so you can easily see which images come from the same plate, but I don't know if this is true for every screen. 
<img width="313" alt="screen shot 2016-03-14 at 10 29 23" src="https://cloud.githubusercontent.com/assets/2150062/13741736/b1ee074e-e9cf-11e5-96d9-7b90efc3aa8e.png">

**Comment by eleanorwilliams** [comment]
Both the gene identifiers and gene symbols are listed so each gene is effectively shown twice.  I think people are more likely to search by gene symbol, so you could show just the gene symbols, but if there is no gene symbol show the gene identifier.

**Comment by joshmoore** [comment]
@eleanorwilliams : here's where we should think about how to inform the plugin (or at least the [developer](@aleksantra-tarkowska)) how to choose between equivalent representations.

**Comment by eleanorwilliams** [comment]
@joshmoore: or allow the user to toggle between two representations depending on what they like to use? 

Could we add a tag in the bulkmap-config file that says display this field as first priority in the Genes tab, this as second if first not available?

**Comment by manics** [comment]
> Could we add a tag in the bulkmap-config file that says display this field as first priority in the Genes tab, this as second if first not available?

I like the general idea, but whereas the bulk-map-config only controls the generation of static map-annotations this effectively means making the web-viewer configurable via a user-defined annotation, which is fine with me- but should we try and spec it out a bit more?

**Comment by gusferguson** [comment]
Agree with @manics that this is all getting a bit thumb-sucky - I certainly can't contribute much to the discussion without much firmer requirements and specs.

**Comment by eleanorwilliams** [comment]
Yes, sure.  I think we need to sit down and think about specific requirements and questions people would be trying to answer with these trees and how it would work it different situations e.g. IDR, person in a lab looking at data from several of their screens/datasets etc.

**Comment by joshmoore** [comment]
Perhaps cultural but I don't understand what *thumb-sucky* here means. For me, what's going on is that the UI representation above has brought up a question ("what is a 'gene' in this context?") which we are working through. Is it one thing or two? How will that be represented? Thinking through these is our attempt to get you the reqs/specs you need.

**Comment by gusferguson** [comment]
> Perhaps cultural but I don't understand what thumb-sucky here means

Formulating on the fly without thinking it through.

**Comment by eleanorwilliams** [comment]
This is somewhat back tracking but it might be useful to think what use cases we are trying to answer with the gene and later phenotype trees.  They are an entry point into the IDR or other sets of data that the users can utilize to find out what data is in the IDR without knowing what all the screens are about.  Ultimately users might be asking:

- when gene X is knocked down what are the phenotypes observed, and under what conditions e.g. only stressed cells?
- where is the protein translated from gene X expressed, and under what conditions e.g. only in dividing cells?
- which genes are associated with phenotype Y

The genes and phenotype trees only go part way to answering these questions as they don't link directly to the answers, but instead get to the phenotypes or genes through the annotations on the images.  E.g.
- tell me all the screens that gene X is in - I might then want to look at images associated with particular screens
- get me all the images associated with gene X - then from this a user could click through them and see what phenotypes are linked to them, or possibly download some kind of table listing images and annotation linked to the images if there are many images. 
- tell me all screens that found the phenotype Y - then as with genes above
- after browsing the phenotypes I see that I'm interested in phenotype Y, get me all the images associated with phenotype Y - then similar to genes, the user might click through them to see what genes are associated with them, and again a table of results for download might be useful. 

Don't know if this is useful/too obvious, but was good for me to get it written out what someone might be trying to do.

**Comment by gusferguson** [comment]
@eleanorwilliams - interesting idea to be able to download a table of hits - presumably with URL links to images found and also to gene identifier sites etc. This would effectively allow saving of the search - albeit on your local machine.

**Comment by eleanorwilliams** [comment]
@gusferguson.  Yes URLs to images and to gene identifier/ontology sites.  Not sure what they would want to do after getting the results - download images/features from particular wells for further analysis? Look at other resources like gene expression/pathway databases?

**Comment by eleanorwilliams** [comment]
Suggestion from Rafael on how we could add some drop downs for filtering.  This is back-of-the-envelope and would need detailed thought, but I wanted to record it here.
![rafaelomerotabinterface](https://cloud.githubusercontent.com/assets/2150062/15993825/2667a114-30ea-11e6-966b-a6a430ab070d.jpg)

**Comment by aleksandra-tarkowska** [comment]
> The genes and phenotype trees only go part way to answering these questions as they don't link directly to the answers, but instead get to the phenotypes or genes through the annotations on the images. 

thanks, I will reorganize the tree to follow all the suggestions
